webapp/sources/rudder/rudder-web/src/main/elm/sources/FileManager/View.elm

```diff
@@ -0,0 +1,214 @@
+  , div [ class "fm-text" ] [ text dir ]
+  , if load
+      then div [ class "fm-loader" ]
+      [ svg [ width "25", height "25" ]
+        [ circle [ cx "50%", cy "50%", r "40%" ] []
+        ]
+      ]
+      else div [] []
+  ]
+
+files : Model -> Html Msg
+files model = div
+  [ class "fm-files"
+  , class <| if model.drag then "fm-drag" else ""
+  , onMouseDown (\x y -> EnvMsg <| MouseDown Nothing x y)
+  , onMouseMove <| EnvMsg << MouseMove
+  , onMouseUp <| EnvMsg << MouseUp Nothing
+  , onDragEnter ShowDrop
+  ]
+  [ div [ class "fm-wrap" ]
+    [ div [ class "fm-fluid" ]
+      <| indexedMap (renderFile model) model.files
+      ++ reverse (map (renderUploading model.progress) (range 0 <| model.filesAmount - 1))
+    ]
+  , if model.showDrop && model.hasWriteRights
+    then div [ class "fm-drop", onDragLeave HideDrop, onDrop GotFiles ] []
+    else div [] []
+  ]
+
+arrowIcon : Msg -> Html Msg
+arrowIcon msg = button [ class "fm-arrow", onClick msg ]
+  [
+    svg
+    [ attribute "height" "24"
+    , viewBox "0 0 24 24"
+    , attribute "width" "24"
+    , attribute "xmlns" "http://www.w3.org/2000/svg"
+    ]
+    [ path [ d "M0 0h24v24H0z", fill "none" ] []
+    , path [ d "M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z", fill "#ffffff" ] []
+    ]
+  ]
+
+back : String -> String
+back route = "/" ++ (join "/" <| withDefault [] <| andThen tail <| tail <| reverse <| split "/" route)
+
+renderUploading : Http.Progress -> Int -> Html Msg
+renderUploading progress i = div [ class "fm-file fm-upload" ]
+  [ div [ class "fm-thumb" ]
+    [ if i == 0 then div [ class "fm-progress", style "width" (toPx <| toFloat <| getSent progress) ] [] else div [] []
+    ]
+  , div [ class "fm-name" ] []
+  ]
+
+getSent : Http.Progress -> Int
+getSent progress = case progress of
+  Sending { size, sent } -> sent * 100 // size
+  _ -> 0
+
+renderFile : Model -> Int -> FileMeta -> Html Msg
+renderFile { api, thumbnailsUrl, dir, selected, clipboardDir, clipboardFiles } i file = div
+  [ id <| "fm-file-" ++ fromInt i, class <| "fm-file"
+      ++ (if member file selected then " fm-selected" else "")
+      ++ (if dir == clipboardDir && member file clipboardFiles then " fm-cut" else "")
+  , title file.name
+  , onMouseDown <| (\x y -> EnvMsg <| MouseDown (Just file) x y)
+  , onMouseUp <| EnvMsg << (MouseUp <| Just file)
+  , onDoubleClick <| if file.type_ == "dir" then EnvMsg <| GetLs <| dir ++ file.name ++ "/" else Download
+  ]
+  [ renderThumb thumbnailsUrl api dir file
+  , div [ class "fm-name" ] [ text file.name ]
+  ]
+
+renderThumb : String -> String -> String -> FileMeta -> Html Msg
+renderThumb thumbApi api dir file = if file.type_ == "dir"
+  then div [ class "fm-thumb" ] [ fileIcon ]
+  else renderFileThumb api thumbApi <| dir ++ file.name
+
+fileIcon : Html Msg
+fileIcon = svg [ attribute "height" "48", viewBox "0 0 24 24", attribute "width" "48", attribute "xmlns" "http://www.w3.org/2000/svg" ]
```

Comment:
cc @RaphaelGauthier - if one day you need to draw svg by hand in elm, it's own it's done

Reply:
You won't have to keep an svg (and i hope you will change it @RaphaelGauthier !!)